frame/staking/Cargo.toml

```diff
@@ -32,10 +32,9 @@ sp-application-crypto = { version = "7.0.0", default-features = false, path = ".
 frame-election-provider-support = { version = "4.0.0-dev", default-features = false, path = "../election-provider-support" }
 log = { version = "0.4.17", default-features = false }
 
-# optional dependencies for cargo features
+# Optional imports for benchmarking
 frame-benchmarking = { version = "4.0.0-dev", default-features = false, path = "../benchmarking", optional = true }
 rand_chacha = { version = "0.2", default-features = false, optional = true }
-pallet-bags-list = { default-features = false, optional = true, path = "../bags-list" }
```

Comment:
`pallet_bags_list` is used in https://github.com/paritytech/substrate/blob/9772209ee962fe36c3502afb638e1bd835c23d9e/frame/staking/src/mock.rs

Reply:
Yes I know, but that is only compiled for tests and we have bags list in the dev dependencies.

Reply:
This is the problem you'll end up with upon running `cargo publish`

```
error: failed to verify package tarball
Caused by:
  failed to parse manifest at `pallet-staking-4.0.0/Cargo.toml`
Caused by:
  feature `fuzz` includes `pallet-bags-list/fuzz`, but `pallet-bags-list` is not a dependency
```

The problem is, as mentioned in https://github.com/paritytech/substrate/pull/12899#issuecomment-1346075663, that this dependency is used in features, so after stripping it `cargo` can no longer find it for the features.

Reply:
Yeah had overseen this, fixed it now;